ts/src/core/query/custom_clause_query.ts

```diff
@@ -110,10 +115,15 @@ export class CustomClauseQuery<
       fieldOptions: options.loadEntOptions,
     });
     this.clause = getClause(options);
+    this.source = options.source;
+  }
+
+  __setSource(src: TSource) {
+    this.source = src;
   }
 
   async sourceEnt(_id: ID) {
-    return null;
+    return this.source ?? null;
```

Comment:
Does this not need to fetch any data? What's the point of passing in the ID if it's always the source from the constructor? My assumption was that it was being reused somehow so that each ent would get this same class...or something? Was this just a legacy API from when it did need to be fetched?

Reply:
Needs to fetch data if used based on id. The source ent is used for privacy checks. In this case, we know what it's visible since you already have an instance to pass to it and don't care, I only used it here for correctness. Can add a comment.

Reply:
Ah I see. Yeah, if you don't mind adding a comment because I could see coming back to this sometime in the future and wondering if it's a bug.